lib/linter/linter.js

```diff
@@ -213,6 +213,7 @@ function addDeclaredGlobals(globalScope, configGlobals, { exportedVariables, ena
 
         if (variable) {
             variable.eslintUsed = true;
+            variable.eslintExported = true;
```

Comment:
Here is a new property `variable.eslintExported`.
This represents the variable specified by the `/* exported variableName */` comment (unlike `eslintUsed`,it represents export only).

I'm not sure if this is a good. Is there a better way?

Reply:
I think this is the right approach. `variable.eslintUsed` could have been set to `true` by another rule via [`context.markVariableAsUsed()`](https://github.com/eslint/eslint/blob/131e646e227b9aca3937fe287343bf2c3df408af/lib/linter/linter.js#L1027), so it's not a reliable indicator that the variable is intended to be used by other scripts, which is what we need in this rule and what `/* exported */` means.

But, this is a change in the core so I'd like more opinions from @eslint/eslint-tsc

Reply:
Makes sense to me.